### PR TITLE
Fix EZP-23866: Move legacy_mode setting to LegacyBundle

### PR DESCRIPTION
**Pull request by lolautruche**
> https://jira.ez.no/browse/EZP-23866

* `legacy_mode` semantic setting has been moved to LegacyBundle (under
`ez_publish_legacy` namespace).
* `legacy_aware_routes` semantic setting has also been moved to
* LegacyBundle.
* DIC extensions has been adapted
* Services relying on legacy mode have been adapted

1st commit is to be backported to 5.4 (no feature removed, only deprecated)

### CODE REVIEW

**Comment by lolautruche** [comment]
Review ping @bdunogier @yannickroger @andrerom

**Comment by patrickallaert** [comment]
:+1:

**Comment by yannickroger** [comment]
+1

File: eZ/Bundle/EzPublishLegacyBundle/Tests/Routing/DefaultRouterTest.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * This file is part of the eZ Publish Kernel package.
```
Comment (bdunogier) [inline eZ/Bundle/EzPublishLegacyBundle/Tests/Routing/DefaultRouterTest.php]: Actually not (valid in my code as well). But we'd better do one cleanup PR for this.

**Comment by bdunogier** [comment]
Looks really good !

If Behat agrees, and if it works, +1 for me.

**Comment by lolautruche** [comment]
@bdunogier Yes we asked Behat :wink: . See https://github.com/ezsystems/ezpublish-community/pull/222

**Pull request merged**